<launch>
  <arg name="name" default="quadrotor"/>
  <arg name="type" default="hummingbird"/>
  <arg name="mass" default="0.5"/>
  <arg name="world_frame_id" default="simulator"/>
  <arg name="rate/odom" default="100"/>

  <arg name="initial_position/x" default="0.0"/>
  <arg name="initial_position/y" default="0.0"/>
  <arg name="initial_position/z" default="0.0"/>
  <arg name="initial_orientation/x" default="0.0"/>
  <arg name="initial_orientation/y" default="0.0"/>
  <arg name="initial_orientation/z" default="0.0"/>
  <arg name="initial_orientation/w" default="1.0"/>
  <arg name="color/r" default="0.0"/>
  <arg name="color/g" default="0.0"/>
  <arg name="color/b" default="1.0"/>
  <arg name="color/a" default="0.7"/>

  <node pkg="quadrotor_simulator"
    type="quadrotor_simulator_so3"
    name="quadrotor_simulator_so3"
    output="screen">
    <rosparam file="$(find quadrotor_simulator)/config/$(arg type)_params.yaml"/>
    <param name="world_frame_id" value="$(arg world_frame_id)"/>
    <param name="quadrotor_name" value="$(arg name)"/>
    <param name="rate/odom" value="$(arg rate/odom)"/>
    <param name="mass" value="$(arg mass)"/>
    <remap from="~odom" to="odom"/>
    <remap from="~imu" to="imu"/>
    <remap from="~cmd" to="so3_cmd"/>
    <param name="initial_position/x" value="$(arg initial_position/x)"/>
    <param name="initial_position/y" value="$(arg initial_position/y)"/>
    <param name="initial_position/z" value="$(arg initial_position/z)"/>
    <param name="initial_orientation/x" value="$(arg initial_orientation/x)"/>
    <param name="initial_orientation/y" value="$(arg initial_orientation/y)"/>
    <param name="initial_orientation/z" value="$(arg initial_orientation/z)"/>
    <param name="initial_orientation/w" value="$(arg initial_orientation/w)"/>
  </node>

  <node pkg="mesh_visualization"
    type="mesh_visualization"
    name="mesh_visualization"
    output="screen">
    <param name="mesh_resource" value="package://mesh_visualization/mesh/hummingbird.mesh"/>
    <param name="color/r" value="$(arg color/r)"/>
    <param name="color/g" value="$(arg color/g)"/>
    <param name="color/b" value="$(arg color/b)"/>
    <param name="color/a" value="$(arg color/a)"/>
    <remap from="~input" to="odom"/>
  </node>
</launch>
